azure-pipelines/build/doclient-lite-docker.yml

```diff
@@ -1,6 +1,10 @@
 # Pipeline to build DO Agent using docker to target non-native OS and/or architecture.
 # Publishes the binaries + packages as artifacts.
 
+variables:
+- name: version
```

Comment:
>version [](http://example.com/codeflow?start=8&length=7)

Do we need this variable? Wouldn't we just update the version in a branch when it needs to be changed?

Reply:
Looks like github comment isnt showing up in codeflow:

It's needed in order to specify which docker image version to use. I can't think of a way to extract version #58 
With this approach, the latest development branch will always have this version set to the next release version

Reply:
I meant can't we hardcode it in the yaml file itself? When will we need to update the version on the fly just before scheduling a pipeline run?

Reply:
For the case where we need to patch a previous release. For example if we needed to patch v0.7.0 which uses cpprest - we can now specify it through the pipeline to use this set of docker images to do our builds.

Reply:
Okay consider renaming to something more specific. vmImageVersion?

Reply:
maybe dependencyVersion. it might be used for both the docker image version & for specifying the image of the vmImage within the pool

Reply:
buildEnvVersion?

Reply:
i think imageVersion is fine - referring to both vm & docker images. Added a comment next to it as well.

Waiting on the PR completion from ImageFactory and seeing runs go through before checking this in.

Reply:
PR is completed, there appears to be an issue with updating the image payload within cloudtest - as a result, the reported new do-adu-build-0.8.0 image does not have libcurl installed, despite imagefactory logs showing libcurl being installed during image provisioning.

I'm going to complete the PR for now, but we won't see green builds until we get a response from the imagefactory team.